Demo/Demo/ViewModels/PayPalVaultViewModel.swift

```diff
@@ -16,8 +16,15 @@ class PayPalVaultViewModel: VaultViewModel {
             let vaultRequest = PayPalVaultRequest(setupTokenID: setupTokenID)
             paypalClient.vault(vaultRequest) { result, error in
```

Comment:
qq: For the `PayPalClient.vault()` callback can do `error: CoreSDKError` instead of `error: Error`?

Reply:
I followed the Braintree example where merchant doesn't have to know about error types unless they need to for specific purposes.

Reply:
The reason for opening this PR was while I was working on migration guide, if the merchant wants to treat cancel error separately (I've seen these asks coming in from merchants on Braintree side), it is quite cumbersome the way we have the errors set up in our iOS SDK. They have to cast to CoreSDKError, then look for specific number for code and value for domain. I thought about exposing CardClientError as public so they can check those values CardClientError.domain, but that seemed very messy too because CardClientError internally just creates CoreSDKError.

Reply:
I see your point, so that the merchant is aware of type of errors we are throwing.

Reply:
Yeah to be honest the type wrangling seems unnecessary since we could just return `CoreSDKError`. It results in less code and feels a lot cleaner. Would it be worth it to consider diverging from the BT way?

Reply:
they'd still have to check for code an domain, I guess I can make CardClientError and PayPalWebCheckoutError public so they can check CardClientError.domain, etc.

Reply:
I was thinking also, if we do restructuring of the errors, I think we should eventually,
this helper function abstracts away the implementation

Reply:
Yeah honestly the domain portion feels cumbersome too. It feels like it's only there because the `Error` type is so generic. I'd say go what feels best but I wouldn't get tied down into how BT does it if there's a better way.

It's hard for me to comment from the Android side. We're using sealed classes so we don't need `domain` and `errorCode`. Kotlin's type system captures all of the error type information. Merchants can use `switch` statements to get more detail on the exact error cause.

Reply:
Yes, I initially just wrote that way on migration guide with switch statement and then I tried it. ❌